src/syntax_mapping.rs

```diff
@@ -81,6 +81,16 @@ impl<'a> SyntaxMapping<'a> {
                 .unwrap();
         }
 
+        // Support syntax highlighting for Tmux, resolves issue #1703
+        for glob in &[
+            "tmux.conf",
+	    "tmux",
+	    ".tmux.conf",
+	    ".tmux",
```

Comment:
```suggestion
            "tmux.conf",
            "tmux",
            ".tmux.conf",
            ".tmux",
```

Reply:
Thanks a lot for looking into this issue!

I think it might be possible to solve this in a generic way, i.e. not only for tmux.conf.

Our current algorithm to detect syntax is this:

1. See if there is a syntax mapping. If there is a syntax mapping to "unknown", use first-line fallback (this is the code path wrongly taken for tmux.conf). If there is a syntax mapping to a specific syntax, use that syntax
2. See if any syntax handles the file name in question, e.g. `.tmux.conf`. This is the code path we want for `.tmux.conf`.
3. See if any syntax handles the file name extension in question.
4. If all else fails, use first_line fallback

The code for this logic is here: https://github.com/sharkdp/bat/blob/master/src/assets.rs#L214

I wonder if it wouldn't be possible to introduce the concept of "map to unknown syntax, but only if there is no exact file name match". So that instead of

```
        mapping
            .insert("*.conf", MappingTarget::MapToUnknown)
            .unwrap();
```

we would have something like

```        mapping
            .insert("*.conf", MappingTarget::MapToUnknownUnlessExactFileNameMatch)
            .unwrap();
```

That would also allow us to get rid of this special case:

```
        // Re-insert a mapping for resolv.conf, see #1510
        mapping
            .insert("resolv.conf", MappingTarget::MapTo("resolv"))
            .unwrap();
```

Reply:
Thank you for the suggestion. I can look into making the change.